lib/webhdfs/client_v1.rb

```diff
@@ -408,5 +572,34 @@ def request(host, port, method, path, op=nil, params={}, payload=nil, header=nil
         end
       end
     end
+
+    private
+    def smart_request(method, path, op, params={}, payload=nil)
```

Comment:
Similar question to above -- was this stuff ported in from NIM or something? Curious how we figured out all ins and outs of this error handling.

Reply:
Implemented in 8cf5f5c